module/os/freebsd/spl/spl_taskq.c

```diff
@@ -124,6 +123,28 @@ system_taskq_fini(void *arg)
 SYSUNINIT(system_taskq_fini, SI_SUB_CONFIGURE, SI_ORDER_ANY, system_taskq_fini,
     NULL);
 
+#ifdef __LP64__
+static taskqid_t
+__taskq_genid(void)
+{
+
+	return (atomic_fetchadd_long(&tqidnext, 1) + 1);
```

Comment:
Extra +1?

Reply:
This is the idiomatic use, you can grep the tree to check. Semantics of the routine are "add 1, but return whatever was there prior to it", meaning patched code uses the very id it bumped to, while original code uses the previous one.

Reply:
In that case perhaps we should be using `atomic_add_64_nv` from `sys/atomic.h` (in spl).
```C
static inline uint64_t
atomic_add_64_nv(volatile uint64_t *target, int64_t delta)
{
        return (atomic_fetchadd_64(target, delta) + delta);
}
```